libraries/botframework-connector/src/auth/passwordServiceClientCredentialFactory.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as adal from 'adal-node';
+import * as t from 'runtypes';
+import { ServiceClientCredentials } from '@azure/ms-rest-js';
+import { Maybe, tests } from 'botbuilder-stdlib';
+import { MicrosoftAppCredentials } from './microsoftAppCredentials';
+import { ServiceClientCredentialsFactory } from './serviceClientCredentialsFactory';
+import { AuthenticationConstants } from './authenticationConstants';
+import { GovernmentConstants } from './governmentConstants';
+
+// Implementation of string.IsNullOrEmpty(): https://docs.microsoft.com/en-us/dotnet/api/system.string.isnullorempty?view=netcore-3.1
+const stringIsNullOrEmpty = (val: unknown): val is Maybe<string> => {
+    return tests.isNil(val) || (tests.isString(val) && !val.length);
+};
+
+/**
+ * A simple implementation of the [ServiceClientCredentialsFactory](xref:botframework-connector.ServiceClientCredentialsFactory) interface.
+ */
+export class PasswordServiceClientCredentialFactory implements ServiceClientCredentialsFactory {
+    /**
+     * The app ID for this credential.
+     */
+    appId: string | null;
+
+    /**
+     * The app password for this credential.
+     */
+    password: string | null;
+
+    constructor(appId: string, password: string) {
+        this.appId = appId;
+        this.password = password;
+    }
+
+    async isValidAppId(appId: string): Promise<boolean> {
+        return appId === this.appId;
+    }
+
+    async isAuthenticationDisabled(): Promise<boolean> {
+        return stringIsNullOrEmpty(this.appId);
+    }
+
+    async createCredentials(
+        appId: string,
+        audience: string,
+        loginEndpoint: string,
+        validateAuthority: boolean
+    ): Promise<ServiceClientCredentials> {
+        t.Record({
+            appId: t.String.withConstraint((s) => s === this.appId),
```

Comment:
Good question! That's because the `ServiceClientCredentialFactory` is a replacement for the `ICredentialProvider` which supports multiple bots (practically speaking, multiple AAD AppIds/AppPasswords/Certificates)